base/sysinfo.jl

```diff
@@ -431,23 +431,33 @@ windows_version
 
 const WINDOWS_VISTA_VER = v"6.0"
 
-"""
-    Sys.isexecutable(path::String)
 
-Return `true` if the given `path` has executable permissions.
-"""
-function isexecutable(path::String)
-    if iswindows()
-        return isfile(path)
```

Comment:
@staticfloat  I suppose this is what you meant when you said it is unreliable, afterall it simply checks if the path is a file.